src/catalog.js

```diff
@@ -181,15 +181,17 @@ angular.module('gettext').factory('gettextCatalog', function (gettextPlurals, ge
                     val = obj;
                 }
 
+                this.strings[language][key] = this.strings[language][key] || {};
                 // Expand single strings for each context.
                 for (var context in val) {
                     var str = val[context];
                     if (!angular.isArray(str)) {
-                        val[context] = [];
-                        val[context][defaultPlural] = str;
+                        this.strings[language][key][context] = [];
```

Comment:
Not if the original `this.strings[language][key][context]` already have some value. As per test:
```
        catalog.setStrings("en", { "a million dollar": { rich_context: "nothing" } });
        catalog.setStrings("en", { "a million dollar": { poor_context: "dream" } });
```
the second call overwrites the first one, even so they are not in the same context, because the `val` does not know anything what is already in the `this.strings[language][key][context]`